packages/ui5-application-writer/templates/optional/loadReuseLibs/webapp/test/locate-reuse-libs.js

```diff
@@ -100,45 +102,63 @@
             if (libs && libs.length > 0) {
                 var url = "/sap/bc/ui2/app_index/ui5_app_info?id=" + libs;
-                var sapClient = jQuery.sap.getUriParameters().get("sap-client");
-                if (sapClient && sapClient.length === 3) {
-                    url = url + "&sap-client=" + sapClient;
-                }
-                return $.ajax(url).done(updateModuleDefinition);
+                var sapClient = "";
+
+                return new Promise(
+                    function (resolve) {
+                        sap.ui.require(["sap/base/util/UriParameters"], function (UriParameters) {
+                            sapClient = UriParameters.fromQuery(window.location.search).get("sap-client");
+                            if (sapClient && sapClient.length === 3) {
+                                url = url + "&sap-client=" + sapClient;
+                            }
+                            resolve(url);
+                        });
+                    }).then(function (url2) {
+                        return $.ajax(url2).done(function (data) {
+                            if (data) {
+                                registerModules(data);
+                            }
+                        });
+                    });
+            } else {
+                return undefined;
             }
         });
     };
 })(sap);
 
 /*eslint-disable sap-browser-api-warning, sap-no-dom-access*/
-var scripts = document.getElementsByTagName("script");
 var currentScript = document.getElementById("locate-reuse-libs");
 if (!currentScript) {
     currentScript = document.currentScript;
 }
 var manifestUri = currentScript.getAttribute("data-sap-ui-manifest-uri");
 var componentName = currentScript.getAttribute("data-sap-ui-componentName");
 var useMockserver = currentScript.getAttribute("data-sap-ui-use-mockserver");
-var bundleResources = function () {
-    jQuery.sap.require("jquery.sap.resources");
-    var sLocale = sap.ui.getCore().getConfiguration().getLanguage();
-    var oBundle = jQuery.sap.resources({
-        url: "i18n/i18n.properties",
-        locale: sLocale
-    });
-    document.title = oBundle.getText("appTitle");
-};
+
 sap.registerComponentDependencyPaths(manifestUri)
     .catch(function (error) {
-        jQuery.sap.log.error(error);
+        sap.ui.require(["sap/base/Log"], function (Log) {
+            Log.error(error);
+        });
     })
     .finally(function () {
-        // setting the app title with internationalization
-        sap.ui.getCore().attachInit(bundleResources);
+
+        // setting the app title with internationalization 
+        sap.ui.getCore().attachInit(function () {
+            var sLocale = sap.ui.getCore().getConfiguration().getLanguage();
+            sap.ui.require(["sap/base/i18n/ResourceBundle"], function (ResourceBundle) {
+                var oResourceBundle = ResourceBundle.create({
+                    url: "i18n/i18n.properties",
```

Comment:
Typically, the resource bundles will be named `i18n/i18n.*properties`, but my understanding is that they don't _have_ to follow the naming convention. The authoritative place for the name would be the manifest. Do we need to handle the case where the resource bundle can't be located?

Reply:
If they change the location/name, they'll have to update this file as well. Fair enough